### PR TITLE
Unwrap event recursively from SNS & SQS

### PR DESCRIPTION
**Pull request by nimirium**
It's time for us to feel chill about adding pipeline tools (SNS, SQS) before lambdas, without changing the lambda code.

### CODE REVIEW

File: sosw/components/helpers.py [hunk 1]
```diff
@@ -916,8 +927,114 @@ def is_event_from_sns(event):
...
+        unwrapped_messages = unwrap_event_recursively(event, sources=['sns', 'sqs']):
+
+    :param event: Lambda event
+    :param sources: List of strings describing what the event might be wrapped by. If empty, will unwrapped from all.
+    :return: List of dictionaries - unwrapped messages from the event
+    """
+
+    messages = [event]
+    sources = sources or ['sns', 'sqs']
+    sources = [x.lower() for x in sources]
```
Comment (markbulgakov) [inline sosw/components/helpers.py]: Can be 1 liner
sources = [x.lower() for x in sources or ['sns', 'sqs']]

File: sosw/components/helpers.py [hunk 2]
```diff
@@ -916,8 +927,114 @@ def is_event_from_sns(event):
...
+                    unwrapped = [msg]
+
+                converted_messages.extend(unwrapped)
+
+            messages = converted_messages
+
+        if original == messages:
+            break
+
+    return messages if len(messages) > 1 else messages[0]
```
Comment (markbulgakov) [inline sosw/components/helpers.py]: Why returns a list of msgs or single msg?

File: sosw/components/helpers.py [hunk 3]
```diff
@@ -892,6 +896,13 @@ def to_bool(val):
     raise Exception(f"Can't convert unexpected value to bool: {val}, type: {type(val)}")
 
 
+def _unwrap_msg_dict_from_sns_event(event) -> Dict:
+    try:
+        return json.loads(event['Records'][0]['Sns']['Message'])
```
Comment (ngr) [inline sosw/components/helpers.py]: I suggest to add a validation meanwhile that there is ONLY ONE record in the event.
At some point we shall face some batches of SNS messages coming in same event to the Lambda. We better notice this immediately.

Comment (nimirium) [reply sosw/components/helpers.py]: added

**Pull request merged**